### PR TITLE
Fixed the Null Pointer Exception on mdm-submit operation

### PR DESCRIPTION
**Pull request by Qingyixia**
#4750

### CODE REVIEW

File: hapi-fhir-storage-mdm/src/main/java/ca/uhn/fhir/mdm/batch2/submit/MdmInflateAndSubmitResourcesStep.java [hunk 1]
```diff
@@ -53,6 +53,7 @@ public class MdmInflateAndSubmitResourcesStep implements IJobStepWorker<MdmSubmi
 	private ResponseTerminologyTranslationSvc myResponseTerminologyTranslationSvc;
 	@Autowired
 	private IMdmChannelSubmitterSvc myMdmChannelSubmitterSvc;
+	@Autowired
```
Comment (lukedegruchy) [inline hapi-fhir-storage-mdm/src/main/java/ca/uhn/fhir/mdm/batch2/submit/MdmInflateAndSubmitResourcesStep.java:56]: I'm guessing this was either due to a bad merge or something missing in a previous ticket?

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_6_0/4750-null-pointer-exception-on-mdm-submit.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+type: fix
+issue: 4750
+jira: SMILE-6152
+title: "Previously, mdm-submit operation with Prefer: response-async header will result in error logs of NullPointerException
```
Comment (lukedegruchy) [inline hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_6_0/4750-null-pointer-exception-on-mdm-submit.yaml:5]: Suggestion:

"Previously, an mdm-submit operation with Prefer: response-async header resulted in a NullPointerException in the error logs
and a batch job status of FAILED.  This has been fixed."

**Review by lukedegruchy — APPROVED** [review]

File: hapi-fhir-storage-mdm/src/main/java/ca/uhn/fhir/mdm/batch2/submit/MdmInflateAndSubmitResourcesStep.java (hunk 1 above)
Comment (Qingyixia) [reply hapi-fhir-storage-mdm/src/main/java/ca/uhn/fhir/mdm/batch2/submit/MdmInflateAndSubmitResourcesStep.java:56]: Yeah, someone missed the '@Autowired' when adding the field _myIdHelperService_ which causes the NullPointerException.

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_6_0/4750-null-pointer-exception-on-mdm-submit.yaml (hunk 2 above)
Comment (Qingyixia) [reply hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_6_0/4750-null-pointer-exception-on-mdm-submit.yaml:5]: Addressed, thx!

**Pull request merged**